python/python/tests/test_dataset.py

```diff
@@ -384,6 +384,38 @@ def adder(batch: pa.RecordBatch) -> pa.RecordBatch:
     )
 
 
+def test_cleanup_old_versions(tmp_path):
+    table = pa.Table.from_pydict({"a": range(100), "b": range(100)})
+    base_dir = tmp_path / "test"
+    lance.write_dataset(table, base_dir)
+
+    dataset = lance.dataset(base_dir)
+    # These tests don't actually clean up any files.  Due to the 2 weeks
+    # requirement this is currently too difficult to simulate from python.
+    # Instead we just make sure the API is correct.
```

Comment:
Yeah I think testing is another case where an override parameter might be good.

https://github.com/lancedb/lance/issues/1269#issuecomment-1714848695